.github/workflows/build.yaml

```diff
@@ -22,34 +22,47 @@ concurrency:
 # will skip running `yarn install` if it successfully fetched from cache
 
 jobs:
-  fmt:
+  prettier:
     name: Format with Prettier
     runs-on: ubuntu-latest
     timeout-minutes: 5
     steps:
       - name: Checkout repo
         uses: actions/checkout@v3
 
+      - name: Run prettier with actionsx/prettier
+        uses: actionsx/prettier@v2
+        with:
+          args: --check --loglevel=warn .
+
+  doctoc:
+    name: Doctoc markdown files
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+
+      - name: Get changed files
+        id: changed-files
+        uses: tj-actions/changed-files@v26.1
+        with:
+          files: |
+            docs/**
+
       - name: Install Node.js v16
+        if: steps.changed-files.outputs.any_changed == 'true'
         uses: actions/setup-node@v3
         with:
           node-version: "16"
+          cache: "yarn"
 
-      - name: Fetch dependencies from cache
-        id: cache-node-modules
-        uses: actions/cache@v3
-        with:
-          path: "**/node_modules"
-          key: yarn-build-${{ hashFiles('**/yarn.lock') }}
-          restore-keys: |
-            yarn-build-
-
-      - name: Install dependencies
-        if: steps.cache-node-modules.outputs.cache-hit != 'true'
-        run: SKIP_SUBMODULE_DEPS=1 yarn --frozen-lockfile
+      - name: Install doctoc
+        run: yarn global add doctoc@2.0.0
```

Comment:
Didn't want to have to install all the modules to use `doctoc` so went with this approach.

Reply:
Similar to prettier we might one day want to make sure this uses the pinned version in the lockfile (or pin both) to avoid desync between CI and local.